### PR TITLE
chore: update some code and docs to use v9 API

### PR DESCRIPTION
**Pull request by zylalx1**
### Description of the feature or fix

update most of the remaining code (including demos, examples and tests) and docs using the old API in lv_api_map_v8.h to v9 API

i keep the code using lv_obj_move_foreground and lv_obj_move_background untouched because i think these APIs are nice and easy to use

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.10](https://github.com/szepeviktor/astyle/releases/tag/v3.4.10) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: demos/music/assets/img_lv_demo_music_cover_1.c [hunk 1]
```diff
@@ -184,7 +184,7 @@ const LV_ATTRIBUTE_MEM_ALIGN uint8_t img_lv_demo_music_cover_1_map[] = {
...
 };
 
-lv_img_dsc_t img_lv_demo_music_cover_1 = {
+lv_image_dsc_t img_lv_demo_music_cover_1 = {
```
Comment (XuNeo) [inline demos/music/assets/img_lv_demo_music_cover_1.c]: It should be a const var. Let's correct it.

```suggestion
const lv_image_dsc_t img_lv_demo_music_cover_1 = {
```

File: demos/music/assets/img_lv_demo_music_cover_2.c [hunk 2]
```diff
@@ -185,7 +185,7 @@ const LV_ATTRIBUTE_MEM_ALIGN uint8_t img_lv_demo_music_cover_2_map[] = {
 
 };
 
-lv_img_dsc_t img_lv_demo_music_cover_2 = {
+lv_image_dsc_t img_lv_demo_music_cover_2 = {
```
Comment (XuNeo) [inline demos/music/assets/img_lv_demo_music_cover_2.c]:
```suggestion
const lv_image_dsc_t img_lv_demo_music_cover_2 = {
```

File: demos/music/assets/img_lv_demo_music_cover_3.c [hunk 3]
```diff
@@ -185,7 +185,7 @@ const LV_ATTRIBUTE_MEM_ALIGN uint8_t img_lv_demo_music_cover_3_map[] = {
 
 };
 
-lv_img_dsc_t img_lv_demo_music_cover_3 = {
+lv_image_dsc_t img_lv_demo_music_cover_3 = {
```
Comment (XuNeo) [inline demos/music/assets/img_lv_demo_music_cover_3.c]:
```suggestion
const lv_image_dsc_t img_lv_demo_music_cover_3 = {
```

**Review by kisvegabor — COMMENTED** [review]
Very nice, thank you so much!

Please the fix the missing `const`s pointed out by @XuNeo and we are good to go from my side!

**Review by XuNeo — DISMISSED** [review]

**Review by FASTSHIFT — DISMISSED** [review]

**Comment by kisvegabor** [comment]
I've fixed the merge conflicts.

**Review by XuNeo — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**